src/main/java/com/karankumar/bookproject/backend/service/CustomShelfService.java

```diff
@@ -48,15 +48,15 @@ public CustomShelf createCustomShelf(String shelfName) {
         return new CustomShelf(shelfName, userService.getCurrentUser());
     }
 
-    public CustomShelf findById(Long id) {
-        return customShelfRepository.getOne(id);
+    public Optional<CustomShelf> findById(Long id) {
+        return Optional.ofNullable(customShelfRepository.getOne(id));
```

Comment:
If we change this to `findById()`, then it will fail the `deleteCustomShelf` test in `CustomShelfServiceTest.java`. It seems that the test method expects an exception to be thrown, which does not occur after the change.

Reply:
Could you please change the test so that it passes? You may only need to check whether the ID coming back is empty